src/v0/destinations/af/transform.js

```diff
@@ -130,28 +130,29 @@ function getEventValueForUnIdentifiedTrackEvent(message) {
 
 function getEventValueMapFromMappingJson(message, mappingJson, isMultiSupport, config) {
   let eventValue = {};
-  const { addPropertiesAtRoot, afCurrencyAtRoot } = config;
-
+  const { addPropertiesAtRoot, afCurrencyAtRoot, listOfProps } = config;
+  const clonedProp = message.properties && JSON.parse(JSON.stringify(message.properties));
   if (addPropertiesAtRoot) {
-    eventValue = message.properties;
+    eventValue = clonedProp;
   } else {
-    set(eventValue, 'properties', message.properties);
+    if (Array.isArray(listOfProps) && listOfProps.length > 0) {
+      listOfProps.forEach((prop) => {
```

Comment:
if prop.property is simple string not containing any dots then why use set function?
instead we can set using eventValue[prop.property] = message.properties[properties] or we can use lodash pick method and pass array

Reply:
addressed